### PR TITLE
Add missing 'getVisitorKeys' method type definition for 'Printer'

### LINKED ISSUE #15121: `getVisitorKeys` not part of type definitions
`getVisitorKeys` is not part of the Prettier type definitions.

**Environments:**

- Prettier Version: 3.0.0
- Running Prettier via: Node
- Runtime: Node 16
- Operating System: Windows

**Steps to reproduce:**

```ts
import { Printer } from 'prettier';

export const printers: Record<string, Printer> = {
    'my-printer': {
        print: null as any,
        embed: null as any,
        getVisitorKeys: null as any,
    },
};
```

**Expected behavior:**

`getVisitorKeys` should be part of the type definitions

**Actual behavior:**

Code snippet errors because `getVisitorKeys` is missing

### PR DESCRIPTION
**Pull request by auvred**
## Description

Closes #15121

Reference: https://prettier.io/docs/en/plugins.html#optional-getvisitorkeys

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Can you add changelog?

**Comment by auvred** [comment]
I'm not sure how to properly describe this change 🙃

File: changelog_unreleased/api/15125.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#### Add missing 'getVisitorKeys' method type definition for 'Printer' (#15125 by @auvred)
```
Comment (fisker) [inline changelog_unreleased/api/15125.md:1]:
```suggestion
#### Add missing `getVisitorKeys` method type definition for `Printer` (#15125 by @auvred)
```

**Comment by fisker** [comment]
> I'm not sure how to properly describe this change 🙃

I think it's good enough.

**Review by fisker — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**